hail/src/main/scala/is/hail/expr/ir/StagedRegion.scala

```diff
@@ -53,13 +66,41 @@ object StagedRegion {
   }
 }
 
-class RealStagedRegion(r: Value[Region]) extends StagedRegion {
+class RealStagedRegion(r: Value[Region]) extends StagedRegion { self =>
   def code: Value[Region] = r
 
   def createChildRegion(mb: EmitMethodBuilder[_]): StagedOwnedRegion = {
     val newR = mb.newLocal[Region]("staged_region_child")
     new RealStagedOwnedRegion(newR, this)
   }
+
+  def createChildRegionArray(mb: EmitMethodBuilder[_], length: Int): StagedOwnedRegionArray = {
+    val regionArray = mb.newLocal[Array[Region]]("staged_region_child_array")
+
+    def get(i: Value[Int]): Settable[Region] = new Settable[Region] {
+      def get: Code[Region] = regionArray(i)
+
+      def store(rhs: Code[Region]): Code[Unit] = regionArray.update(i, rhs)
+    }
+
+    new StagedOwnedRegionArray {
+      def apply(i: Value[Int]): StagedOwnedRegion = new RealStagedOwnedRegion(get(i), self)
+
+      def allocateRegions(mb: EmitMethodBuilder[_], size: Int): Code[Unit] = {
+        val i = mb.newLocal[Int]("sora_alloc_i")
+        Code(
+          regionArray := Code.newArray(length),
```

Comment:
is type inference filling in the type info here?

Reply:
Yeah, `regionArray` is a `ThisFieldRef[Array[Region]]`, so `regionArray.:=` expects a `Code[Array[Region]]`.